lib/util.js

```diff
@@ -231,11 +231,19 @@ exports.parseParam = function (val, spec) {
   if (spec.fields &&
       spec.fields.subjectLiteral &&
       spec.fields.subjectLiteral.field === 'subjectLiteral_exploded' &&
-      val.subjectLiteral &&
-      val.subjectLiteral.slice(-1) === '.'
+      val.subjectLiteral
     ) {
-    val.subjectLiteral = val.subjectLiteral.slice(0, -1)
-    logger.debug('Removing terminal period', JSON.stringify(val, null, 4))
+    if (typeof val.subjectLiteral === 'string' && val.subjectLiteral.slice(-1) === '.') {
+      val.subjectLiteral = val.subjectLiteral.slice(0, -1)
+      logger.debug('Removing terminal period', JSON.stringify(val, null, 4))
+    } else if (Array.isArray(val.subjectLiteral)) {
+      val.subjectLiteral.forEach((sub, i) => {
+        if (sub.slice(-1) === '.') {
+          val.subjectLiteral[i] = sub.slice(0, -1)
+          logger.debug('Removing terminal period', JSON.stringify(sub, null, 4))
+        }
+      })
+    }
```

Comment:
@danamansana is this change familiar and may it go to production?

Reply:
I remember doing this but I don't know why it didn't make it into production. I will try to find out where it came from. On the face of it it looks unlikely to cause problems though since we are just extending to arrays what we already have in production for strings.

Reply:
I think this was done in the beginning of the subject heading work, in an attempt to normalize the data that SHEP would be ingesting. Before we merge it in I'd prefer to run it by Stephen. If we need to update production today I'd leave this part out.

Reply:
@thisisstephenbetts any thoughts on these changes?

Reply:
Looking...

Reply:
I do remember something about arrays vs strings — but I don't remember why we get both. 

If it never made it into production I suspect this was superseded by other work. 

Actually digging in, it looks like it was a change from 14 months ago, so I'm sure it'd be from the exploded subjects in search thing, not from SHEP. So I would 100% bet it's not needed now.

Reply:
So, it seems this change is [on `production`](https://github.com/NYPL-discovery/discovery-api/compare/production?expand=1), but not master. And I had cut from master and now we are getting a strange and misleading diff.

Reply:
It does appear that this code is on production. I don't understand why git reports it as a difference. When I perform the merge locally, the diff is less concerning:
```
git merge qa
Merge made by the 'recursive' strategy.
 .nvmrc            |    2 +-
 .travis.yml       |    2 +-
 package-lock.json | 1104 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++----------------------------
 package.json      |   14 +-
 4 files changed, 901 insertions(+), 221 deletions(-)
```

[Full diff here](https://gist.github.com/nonword/523d84e421fa5d8a693d494bd4a7bd9c)